lib/schema/src/openctx.schema.ts

```diff
@@ -1,18 +1,66 @@
 /**
  * Metadata about code
  */
-export type Schema = Item | Annotation
+export type Schema = CandidateItem | Item | Annotation
 /**
  * A hint about how to best present an annotation to the human in the client's user interface.
  *
  * - `prefer-link-over-detail`: Prefer to show the item as a link over showing the detail text, if the client does not cleanly support doing both.
  */
 export type PresentationHint = 'prefer-link-over-detail'
+/**
+ * Additional information which can be used to idenfity the resource by the provider.
+ */
+export type ItemMeta = { [key: string]: any }
```

Comment:
What's an example of what this would contain? like the google docs document ID? or the github PR/issue number?

I think we should just rely on the URL for this. Providers can always encode extra stuff in a URL if needed.

I think using the URL suffices for all current known use cases. If we find this ItemMeta is truuuuly needed, we can add it later on.

Reply:
I like the URL idea. My thinking around ItemMeta was there may be work done generating a candidate that you don't want to lose when generating the final item. However, that is indeed theoretical and we should only introduce when needed.

One potential issue with encoding stuff in the URL is we want it to remain clickable. Maybe we should rename URL to URI so we can encode extra stuff in it. Or I suppose if we wanted to encode lots of stuff into the uri you would use a custom url scheme.

Reply:
I have removed the meta for now.